analysis/filter/StructureModifier/src/main/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/NodeRemover.kt

```diff
@@ -0,0 +1,52 @@
+package de.maibornwolff.codecharta.filter.structuremodifier
+
+import de.maibornwolff.codecharta.model.*
+
+class NodeRemover(private val project: Project, private val projectName: String?) {
+
+    fun remove(paths: Array<String>): Project {
+        val pathSegments = paths.map { it.removePrefix("/").removeSuffix("/").split("/") }
+        return ProjectBuilder(
+                projectName ?: project.projectName,
+                removeNodes(pathSegments),
+                removeEdges(paths),
+                copyAttributeTypes(),
+                removeBlacklistItems(paths)
+        ).build()
+    }
+
+    private fun filterNodes(path: List<String>, node: MutableNode): MutableNode {
+        val filteredChildren = node.children.filter { it.name != path.firstOrNull() || path.size > 1 }
+        node.children = filteredChildren.map {
+            filterNodes(path.drop(1), it)
+        }.toMutableList()
+        return node
+    }
+
+    private fun removeNodes(paths: List<List<String>>): MutableList<MutableNode> {
+        var root = project.rootNode.toMutableNode()
+        paths.forEach { root = filterNodes(it.drop(1), root) }
```

Comment:
What do you think about a method called similar to this? Feel free to adjust the method name, but I think it's easier to read.
```
pathWithoutRootName(path) {
    return path.drop(1)
}
```

Reply:
Okey I just realised, the method is used more often... what about `pathWithoutFirstFolder()`?